pandas/core/reshape/pivot.py

```diff
@@ -519,6 +519,8 @@ def pivot(
     # If columns is None we will create a MultiIndex level with None as name
     # which might cause duplicated names because None is the default for
     # level names
+    data = data.copy(deep=False)
+    data.index = data.index.copy()
     data.index.names = [
```

Comment:
Is there an opportunity to defer doing this (and avoiding the copy) if we push this to the `if/else` branches below?

Reply:
Both are shallow copies and should take any time compared to the actual pivot op?

Reply:
That's fair. Might be a cleanup opportunity in the future but ok as it fixes the regression